Enable Microsoft.Extensions.DependencyInjection

maryamariyan opened the pull request:
Enables DI and Options.

maryamariyan commented on src/libraries/Common/tests/Extensions/TestingUtils/Microsoft.AspNetCore.Testing/test/AssemblyFixtureTest.cs:
```diff
@@ -19,7 +19,7 @@ public AssemblyFixtureTest(TestAssemblyFixture assemblyFixture, TestCollectionFi
         public TestAssemblyFixture AssemblyFixture { get; }
         public TestCollectionFixture CollectionFixture { get; }
-
+#if !SKIP_FAILING
```
AssemblyFixtureTests seems unrelated to DI.tests. I think this class will be deleted eventually.

ericstj replied:
If that's the case, then don't include the source file.

maryamariyan commented on src/libraries/Common/tests/Extensions/TestingUtils/Microsoft.AspNetCore.Testing/test/ConditionalFactTest.cs:
```diff
@@ -48,12 +48,14 @@ public void ThisTestMustRunOnCLR()
 #error Target frameworks need to be updated.
 #endif
 
+#if !SKIP_FAILING
```
Same here, we dont need to test ConditionFactTest (unrelated to DI.tests). This is only included because DI.Tests uses some classes under TestingUtils/Microsoft.AspNetCore.Testing

maryamariyan commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Common/src/AssemblyTestLog.cs:
```diff
@@ -1,306 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
This and LoggedTestBase already deleted in @JunTaoLuo 's PR: https://github.com/dotnet/extensions/pull/3040.

I had moved Logging.Tests under Microsoft.Extensions.DependencyInjection/tests/Common/, they need to eventually move to Logging tests I think. that is why I am not including files under Common in the test csproj
I'll be manually applying similar changes and will resolve conflicts with his change when I'm bringing over new PRs later

maryamariyan replied:
Resolved: Moved to Logging tests

maryamariyan commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/ServiceCollectionServiceExtensionsTest.cs:
```diff
@@ -8,7 +8,9 @@
 using Microsoft.Extensions.DependencyInjection.Specification.Fakes;
 using Xunit;
 
+#if !SKIP_FAILING
```
This one should be fixed:

- complains
```
The type or namespace name 'Abstractions' does not exist in the namespace 'Microsoft.Extensions.DependencyInjection' (are you missing an assembly reference?) [C:\CodeHub\runtime\src\libraries\Microsoft.Extensions.DependencyInjection\tests\Microsoft.Extensions.DependencyInjection.Tests.csproj
```

ericstj commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj:
```diff
@@ -2,18 +2,20 @@
+    <Compile Include="*.cs" />
+    <Compile Include="DI.Specification.Tests/Fakes/**/*.cs" />
+    <Compile Include="DI.Specification.Tests/*.cs" />
+    <Compile Include="../../Common/tests/Extensions/TestingUtils/Microsoft.AspNetCore.Testing/**/*.cs" />
```
Let's not include common source by wild-card.  Please select each file and set the link metadata.

ericstj commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj:
```diff
@@ -2,18 +2,20 @@
-    <Reference Include="Microsoft.Extensions.DependencyInjection" />
-    <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
-    <Reference Include="Newtonsoft.Json" />
+    <Compile Include="*.cs" />
```
Is there a reason why we aren't just including all source recursively?  `EnableDefaultItems`?

If there is source in this directory that isn't needed, just delete it.

ericstj commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj:
```diff
@@ -2,18 +2,21 @@
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
-    <RootNamespace>Microsoft.Extensions.DependencyInjection</RootNamespace>
+    <DefineConstants>$(DefineConstants);SKIP_FAILING</DefineConstants>
```
Let's try to avoid this define.

maryamariyan commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/ServiceCollectionServiceExtensionsTest.cs:
```diff
@@ -8,7 +8,8 @@
 using Xunit;
 
-using AbstractionResources = Microsoft.Extensions.DependencyInjection.Abstractions.Resources;
+// TODO: Fix
```
This is the remaining TODO

maryamariyan committed:
Enables DI and Options
- Moves files under DI/tests/Common to Logging/tests/DI.Common as they are Logging tests
- Simplifies and enables DI test csproj
- TODO: Fix test on "ServiceCollectionServiceExtensionsTest.cs"

maryamariyan committed:
TODO: fix

maryamariyan committed:
update ref file

safern commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj:
```diff
@@ -2,18 +2,21 @@
 
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
```
These tests will use the `netstandard2.0` implementation, do we want to add a `NetFrameworkCurrent` TFM to the source project and exclude it from the package?

ericstj replied:
Actually they try to test the `net461` build but if fails to build because of missing references.  We need to exclusion.

safern commented on src/libraries/Microsoft.Extensions.DependencyInjection/tests/Microsoft.Extensions.DependencyInjection.Tests.csproj:
```diff
@@ -2,18 +2,21 @@
-    <Reference Include="Microsoft.Extensions.DependencyInjection" />
-    <Reference Include="Microsoft.Extensions.Logging.Abstractions" />
-    <Reference Include="Newtonsoft.Json" />
+    <Compile Include="..\..\Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\ExceptionAssertions.cs">
```
It seems like this file should/could be converged with the AssertExtensions we have in our Private test utilities.

maryamariyan replied:
- [ ] I'll mark this as todo when I'm cleaning up the unused files. (future PR)

ericstj committed:
Use alias to avoid type conflict

maryamariyan committed:
Cleanup Directory.Build.props and test file

maryamariyan committed:
Remove csproj:
- Files under that directory already contained in:
Microsoft.Extensions.DependencyInjection.Tests.csproj

maryamariyan committed:
Also exclude M.E.Configuration.Functional.Tests.csproj
needs all M.E.Configuration projects referenced

maryamariyan commented:
I see mono failure on the CI:
https://helix.dot.net/api/2019-06-17/jobs/97ba643d-68bb-4f46-81a3-a2094e58b3a3/workitems/Microsoft.Extensions.DependencyInjection.Tests/console

ericstj commented:
This one has a callstack: https://helix.dot.net/api/2019-06-17/jobs/80e913aa-a462-4f20-be32-a4a2e16fc14c/workitems/Microsoft.Extensions.DependencyInjection.Tests/console

ericstj committed:
Build for netfx current & suppress a test on mono

maryamariyan commented on src/mono/netcore/CoreFX.issues_linux.rsp:
Should we have used ActiveIssue instead?
```
[ActiveIssue("https://github.com/dotnet/runtime/issues/33894", TestRuntimes.Mono)]
```

ericstj replied:
We could do that, I was trying to find out how to suppress for mono, if we have an attribute that is better.

steveisok replied:
We are still in the process of removing what we skip from rsp files. We're pretty much down to linux & windows.  Attributes like `[ActiveIssue]` are the way to go.

maryamariyan commented:
resolved conflict by rebase.

maryamariyan committed:
Disable DI test failing on mono

ericstj committed:
Ensure netfx conditions apply to all versions

ericstj committed:
Restore System.ComponentModel.Annotations for more frameworks

ericstj commented:
Mono failure is https://github.com/dotnet/runtime/issues/32377.  I'm seeing this on all recent builds, so I don't think it's related to this change.

davidfowl commented:
👏
